shotover-proxy/src/server.rs

```diff
@@ -211,8 +211,13 @@ impl<C: Codec + 'static> TcpCodecListener<C> {
             // Create the necessary per-connection handler state.
             socket.set_nodelay(true)?;
 
+            let (pushed_messages_tx, pushed_messages_rx) =
+                tokio::sync::mpsc::unbounded_channel::<Messages>();
+
             let mut handler = Handler {
-                chain: self.chain.clone(),
+                chain: self
+                    .chain
+                    .clone_with_pushed_messages_tx(pushed_messages_tx.clone()),
```

Comment:
Should this clone_with_pushed_messages_tx logic be moved into the constructor?
Seems hacky to have this important initialization to occur in some extra method called later on.